QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java

```diff
@@ -48,35 +54,51 @@ public static void setup() {
         Cluck.publish("CAN Elevator Any Fault", BooleanMixing.createDispatch(winchCAN.getDiagnosticChannel(ExtendedMotor.DiagnosticType.ANY_FAULT), updateCAN));
         Cluck.publish("CAN Elevator Bus Voltage Fault", BooleanMixing.createDispatch(winchCAN.getDiagnosticChannel(ExtendedMotor.DiagnosticType.BUS_VOLTAGE_FAULT), updateCAN));
         Cluck.publish("CAN Elevator Temperature Fault", BooleanMixing.createDispatch(winchCAN.getDiagnosticChannel(ExtendedMotor.DiagnosticType.TEMPERATURE_FAULT), updateCAN));
-        
-        raising.setFalseWhen(EventMixing.filterEvent(topLimitSwitch, true, Igneous.globalPeriodic));
-        lowering.setFalseWhen(EventMixing.filterEvent(bottomLimitSwitch, true, Igneous.globalPeriodic));
+
+        BooleanInput limitTop = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(0)), Igneous.globalPeriodic);
+        BooleanInput limitMiddleUpper = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(2)), Igneous.globalPeriodic);
+        BooleanInput limitMiddleLower = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(3)), Igneous.globalPeriodic);
+        BooleanInput limitBottom = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(1)), Igneous.globalPeriodic);
+
+        atTop.setTrueWhen(EventMixing.filterEvent(raising, true, BooleanMixing.onPress(limitTop)));
+        atBottom.setTrueWhen(EventMixing.filterEvent(lowering, true, BooleanMixing.onPress(limitBottom)));
+
+        atTop.setFalseWhen(EventMixing.filterEvent(lowering, true, BooleanMixing.onRelease(limitTop)));
+        atBottom.setFalseWhen(EventMixing.filterEvent(raising, true, BooleanMixing.onRelease(limitBottom)));
+
+        atTop.setFalseWhen(BooleanMixing.onPress(BooleanMixing.orBooleans(limitMiddleUpper, limitMiddleLower, limitBottom)));
+        atBottom.setFalseWhen(BooleanMixing.onPress(BooleanMixing.orBooleans(limitMiddleUpper, limitMiddleLower, limitTop)));
+
+        raising.setFalseWhen(EventMixing.filterEvent(atTop, true, Igneous.globalPeriodic));
+        lowering.setFalseWhen(EventMixing.filterEvent(atBottom, true, Igneous.globalPeriodic));
```

Comment:
You may want to consider cutting up Elevator into multiple methods (or classes) to make it easier to understand.